### PR TITLE
Added optimization-based method by Poncelet

### PR DESCRIPTION
**Pull request by leonardgoeke**
Added the optimization-based method from _Poncelet (2016), Selecting representative days for capturing the implications of integrating intermittent renewables in generation expansion planning problems_ to "clustering" methods of the package.

### CODE REVIEW

File: src/clustering/run_clust.jl [hunk 1]
```diff
@@ -459,3 +461,91 @@ function run_clust_hierarchical_medoid(
...
+    - Time until the solver terminates.
+"""
+function run_clust_poncelet_centroid(
+    data_norm::ClustDataMerged,
+    n_clust::Int,
+    iterations::Int;
+    bins::Int = 20,
+    equal_weight::Bool = false,
+    time_limit::Float64=100.0,
+    ponc_optimizer::Any = nothing)
```
[the remark this answers is not in the document]
Comment (leonardgoeke) [reply src/clustering/run_clust.jl]: Good idea. I've changed the definition accordingly.

File: examples/workflow_example_poncelet.jl [hunk 2]
```diff
@@ -0,0 +1,9 @@
+using TimeSeriesClustering
+
+ts_input_data = load_timeseries_data(:CEP_GER1)
+
+# when using kmedoids-exact, one needs to supply the optimizer. Make sure the optimizer is added through Pkg.add()
+
+using Gurobi
+optimizer = Gurobi.Optimizer
```
[the remark this answers is not in the document]
Comment (leonardgoeke) [reply examples/workflow_example_poncelet.jl]: Any solver that supports `MOI.TimeLimitSec()` should work now. I've extended the example accordingly (and corrected the "using kmedodis-exact" in the comment).

**Comment by holgerteichgraeber** [comment]
@leonardgoeke Thank you for the PR, exciting! I am travelling half of this week and haven't gotten to try it out yet, I hope to get to it in the first half of next week.

**Comment by YoungFaithful** [comment]
@holgerteichgraeber could you have a second look at the pull request?

**Review by YoungFaithful — APPROVED** [review]

**Comment by YoungFaithful** [comment]
The tests fail, because of no Documentation `compat`, which we will fix in the dev branch.

**Pull request merged**